backend/src/main/java/com/hackner/musicompass/service/ArtistService.java

```diff
@@ -0,0 +1,39 @@
+package com.hackner.musicompass.service;
+import com.hackner.musicompass.discogsapi.model.DiscogsArtist;
+import com.hackner.musicompass.discogsapi.model.DiscogsArtistSearchResults;
+import com.hackner.musicompass.model.Artist;
+import com.hackner.musicompass.discogsapi.service.DiscogsArtistApiService;
+import com.hackner.musicompass.secret.DiscogsSecret;
+import lombok.Builder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import java.util.Optional;
+
+@Service
+@Builder
+public class ArtistService {
+    private final DiscogsArtistApiService discogsArtistApiService;
+    private final DiscogsSecret discogsSecret;
+
+    @Autowired
+    public ArtistService(DiscogsArtistApiService discogsArtistApiService, DiscogsSecret discogsSecret) {
+        this.discogsArtistApiService = discogsArtistApiService;
+        this.discogsSecret = discogsSecret;
+    }
+
+    public Optional<Artist> getArtistByArtistName (String artistName) {
+        DiscogsArtistSearchResults DiscogsArtistSearchResults = discogsArtistApiService.getDiscogsArtistByArtistName(artistName, discogsSecret.getDiscogsToken());
+        if (DiscogsArtistSearchResults.getResults().length == 0){
+            return Optional.empty();
+        }
+        DiscogsArtist discogsArtist = DiscogsArtistSearchResults.getResults()[0];
```

Comment:
why are you only getting the first search result here? you could return an `Optional<List<Artist>>` here and show multiple artists in your frontend if applicable

Reply:
I cannot get artists by their name from the API. (I can only request artist info using an API-specific ID.) I have to to make a querry for the artist name and will always get a list as response. I checked a few responses and the first result was good during all checks. I think about using this list later on, but for the MVP I will go with this minimal solution.